### PR TITLE
[ESIMD] Doxygen update V - math.

### PR DESCRIPTION
**Pull request by kbobrovs**
- Update documentation comments
- Hide some unintended public constants

Signed-off-by: Konstantin S Bobrovsky <konstantin.s.bobrovsky@intel.com>

### CODE REVIEW

File: sycl/include/sycl/ext/intel/experimental/esimd/math.hpp [hunk 1]
```diff
@@ -856,13 +887,12 @@ ESIMD_NODEBUG
 // Dot product builtins
 #if defined(ESIMD_GEN7_5) || defined(ESIMD_GEN8) || defined(ESIMD_GEN8_5) ||   \
     defined(ESIMD_GEN9) || defined(ESIMD_GEN9_5)
-
-// FIXME: describe the operation better
+
```
Comment (kbobrovs) [inline sycl/include/sycl/ext/intel/experimental/esimd/math.hpp:890]:
```suggestion
```

File: sycl/include/sycl/ext/intel/experimental/esimd/math.hpp [hunk 2]
```diff
@@ -880,13 +910,11 @@ __ESIMD_API simd<T0, SZ> dp2(simd<T1, SZ> src0, U src1, Sat sat = {}) {
...
 }
-
-// FIXME: describe the operation better
-/// Dot product on groups of 4 elements.
-/// \tparam T0 element type of the returned vector.
-/// \tparam T1 element type of the input vector.
-/// \tparam SZ size of the input and returned vectors. Must be a multiple of 4.
-/// \tparam U type of scalar operand \p src1.
+
/// Dot product on groups of 4 elements.
```
Comment (kbobrovs) [inline sycl/include/sycl/ext/intel/experimental/esimd/math.hpp:913]:
```suggestion
/// Dot product on groups of 4 elements.
```

File: sycl/include/sycl/ext/intel/experimental/esimd/math.hpp [hunk 3]
```diff
@@ -904,13 +932,11 @@ __ESIMD_API simd<T0, SZ> dp3(simd<T1, SZ> src0, U src1, Sat sat = {}) {
...
```
Comment (kbobrovs) [inline sycl/include/sycl/ext/intel/experimental/esimd/math.hpp:935]:
```suggestion
/// Dot product on groups of 4 elements.
```

File: sycl/include/sycl/ext/intel/experimental/esimd/math.hpp [hunk 4]
```diff
@@ -928,13 +954,11 @@ __ESIMD_API simd<T0, SZ> dp4(simd<T1, SZ> src0, U src1, Sat sat = {}) {
...
```
Comment (kbobrovs) [inline sycl/include/sycl/ext/intel/experimental/esimd/math.hpp:957]:
```suggestion
/// Dot product on groups of 4 elements.
```

File: sycl/include/sycl/ext/intel/experimental/esimd/math.hpp [hunk 5]
```diff
@@ -952,13 +976,11 @@ __ESIMD_API simd<T0, SZ> dph(simd<T1, SZ> src0, U src1, Sat sat = {}) {
...
 }
-
-// FIXME: describe the operation better
-/// Linear equation.
-/// \tparam RT element type of the output vector.
-/// \tparam T1 element type of the first input vector \p src0.
-/// \tparam T2 element type of the second input vector \p src1.
-/// \tparam SZ size of the second input vector and returned vectors. Must be a
+
/// Linear equation.
```
Comment (kbobrovs) [inline sycl/include/sycl/ext/intel/experimental/esimd/math.hpp:979]:
```suggestion
/// Linear equation.
```

File: sycl/include/sycl/ext/intel/experimental/esimd/math.hpp [hunk 6]
```diff
@@ -28,16 +28,46 @@ namespace intel {
...
 namespace esimd {
 
+/// @addtogroup sycl_esimd_math
+/// @{
+/// @defgroup sycl_esimd_math_ext Hardware-accelerated math.
+///
+/// This is a group of APIs implementing standard math operations which are also
+/// directly supported by the hardware. Usually the hardware support is a
+/// specific message to the "extended math" GPU "shared function" unit, sent via
+/// the \c send or \c sends instructions. Most of the operations do not conform
```
Comment (kychendev) [inline sycl/include/sycl/ext/intel/experimental/esimd/math.hpp:38]: It's actually done with math instruction.

Comment (kbobrovs) [reply sycl/include/sycl/ext/intel/experimental/esimd/math.hpp:38]: right, fixed

**Pull request merged**